public/app/features/alerting/unified/utils/datasource.ts

```diff
@@ -37,6 +45,53 @@ export function getAlertManagerDataSources() {
     .sort((a, b) => a.name.localeCompare(b.name));
 }
 
+// Used only as a fallback for Alert Group plugin
+export function getAllAlertManagerDataSources(): AlertManagerDataSource[] {
+  return [
+    {
+      name: GRAFANA_RULES_SOURCE_NAME,
+      displayName: 'Grafana',
+      imgUrl: 'public/img/grafana_icon.svg',
+    },
+    ...getAlertManagerDataSources().map<AlertManagerDataSource>((ds) => ({
+      name: ds.name,
+      displayName: ds.name.slice(0, 37),
+      imgUrl: ds.meta.info.logos.small,
+      meta: ds.meta,
+    })),
+  ];
+}
+
+export function getAlertManagerDataSourcesByPermission(
+  permission: 'instance' | 'notification'
+): AlertManagerDataSource[] {
+  const availableDataSources: AlertManagerDataSource[] = [];
+  const permissions = {
+    instance: instancesPermissions.read,
+    notification: notificationsPermissions.read,
+  };
+
+  if (contextSrv.hasPermission(permissions[permission].grafana)) {
+    availableDataSources.push({
+      name: GRAFANA_RULES_SOURCE_NAME,
+      displayName: 'Grafana',
+      imgUrl: 'public/img/grafana_icon.svg',
+    });
+  }
+
+  if (contextSrv.hasPermission(permissions[permission].external)) {
+    const cloudSources = getAlertManagerDataSources().map<AlertManagerDataSource>((ds) => ({
+      name: ds.name,
+      displayName: ds.name.slice(0, 37),
```

Comment:
Suggestion: This is mostly just moved from the alertmanager picker but, should we not investigate if we can fix this with some CSS instead of slicing on an arbitrary length? 🤔